syft/generic/frameworks/types.py

```diff
@@ -19,10 +20,18 @@
     framework_tensors.append(torch.nn.Parameter)
     framework_shapes.append(torch.Size)
 
+    framework_layer_module = torch.nn.Module
```

Comment:
Well, this looks interesting :D

Reply:
There is now a tf.Module that layers inherit from

Reply:
Perfect

Reply:
Only concern is that torch.nn.Modules have quite a bit more functionality by default -- tf.Module is literally just a nested collection of variables & submodules with a name/namescope

Reply:
The only thing I ask from a Module is the ability to return the list of pairs ('name_param', param), do you have this in TF?

Reply:
closest thing would be `tf.Module.variables` and `tf.Module.trainable_variables`, but neither of those would get you what you want directly. I'd probably end up doing `named_params = [(v.name, v) for v in module.trainable_variables]` if I needed something similar despite it not being totally equivalent